src/components/OmouComponents/IdleLogout.js

```diff
@@ -53,98 +54,132 @@ const useStyles = makeStyles((theme) => ({
-    const [modalStyle] = useState(getModalStyle);
-    const [open, setOpen] = useState(false);
+    const [modalPosition] = useState(getModalPosition);
+    const [openModal, setOpenModal] = useState(false);
+    const sessionTimeoutRef = useRef(null);
+
+    const TOKEN_REFRESH = gql`
+        mutation refreshToken($token:String) {
+            refreshToken(token:$token){
+            token
+            refreshExpiresIn
+            }
+        }
+    `;
+
+    const [refreshToken, newToken] = useMutation(TOKEN_REFRESH, 
+        {
+            onCompleted: (data) => {
+                (async () => {
+                    store.dispatch(await setToken(data["refreshToken"]["token"], true));
+                })();
+            }
+        });
 
-   
-    function logoutAfter2Minutes() {
-        return new Promise(resolve => {
-            setTimeout(() => {
-                resolve(logoutAndCloseModal());
-            }, 300000);
+    const resetToken = () => {
+        const token = localStorage.getItem("token");
+        refreshToken({
+            "variables" : {
+                "token": token
+            }
         })
     };
     
+    const refreshTokenAfter15Minutes = () => {
+        return new Promise(resolve => {
+            setInterval(() => {
+                resolve(() => {
+                    const token = localStorage.getItem("token");
+                    refreshToken({
+                        "variables": {
+                            "token": token
+                        }
+                    });
+                });
+            }, 900000);
+        });
+    };
+
+    useEffect(() => {
+        (async () => {
+            await refreshTokenAfter15Minutes();
+        })();
+
+    });
+    
     const logoutAndCloseModal = () => {
         handleClose();
         handleLogout();
-    }
+    };
 
     async function handleOpen() {
-        setOpen(true);
+        setOpenModal(true);
         handleReset();
-        setRemaining(modalTimeout);
-        await logoutAfter2Minutes();
+        setRemainingMsUntilPrompt(modalTimeout);
 
-    };
+        sessionTimeoutRef.current = setTimeout(logoutAndCloseModal, 5 * 1000);
+    }
     
     const handleClose = () => {
-        setOpen(false);
+        clearInterval(sessionTimeoutRef.current);
+        setOpenModal(false);
+        resetToken();
     };
 
     const {
         reset,
-        pause,
-        resume,
         getRemainingTime,
-        getLastActiveTime,
-        getElapsedTime
     } = useIdleTimer({
-        timeout,
-        onActive: handleOnActive,
-        onIdle: handleOnIdle
+        "timeout": idleTimeout
     });
     
     const handleReset = () => reset();
-    const handlePause = () => pause();
-    const handleResume = () => resume();
-
-    useEffect(() => {
-        setRemaining(getRemainingTime())
-        setLastActive(getLastActiveTime())
-        setElapsed(getElapsedTime())
     
+    useEffect(() => {
```

Comment:
I'm really confused what this useEffect does. First it sets the remaining miliseconds until "prompt" to whatever is "get remaining time." Then it sets the interval (of what?) to 1000 ms before it sets remaining miliseconds until prompt to whatever is "get remaining time."

I THINK it's counting down before the session logout prompt, then it updates the countdown every second?  Why is this necessary? 

Refactor the function names to make this more readable.

Reply:
The first line in that function was unecessary - I forgot to delete when playing with the token. 

The setInterval is what actually tracks the countdown - getRemainingTime is the react-idle-timer call. It's at an interval so it isn't being called more than every second since that's as precise as necessary to update the state.

Reply:
what's the difference between this useEffect and the earlier useEffect (line 135)?